pkg/controller/nodelifecycle/scheduler/taint_manager_test.go

```diff
@@ -39,7 +40,7 @@ import (
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
-var timeForControllerToProgress = 500 * time.Millisecond
+var timeForControllerToProgress = 10 * time.Millisecond
```

Comment:
why is this enough? The unit tests might run in a very small container in CI, needing more than 10ms.

Ideally, we should eliminate all sleeps and do Polls

Reply:
It is enough because we do periodic poll in the `verifyPodActions` function. If we eliminate the sleep the tests will pass now. I wanted to leave it though, because I've noticed that if I eliminate the sleep entirely and expect in the asserts `expectDelete=false` then the tests pass, because the delete was not executed yet (but would execute in a couple of ms). So I think leaving it as 10ms is pragmatic cause we would detect such issues when running locally.

Reply:
in TestUpdatePod, there is a sleep without a corresponding verify. I think we should add some verify there and then we can simply eliminate all sleeps. And maybe we can do `wait.Poll` instead of `wait.PollImmediate`

Reply:
Done. However, I still keep the `timeForControllerToProgress`, reverted to the previous value as there are 2 tests which check that the controller does not panic after the last update action. Could eliminate them in this PR or a follow up, cause the main goal of stabilizing the tests is achieved by the additional waits already.